modules/participant_portal/php/view_surveys.class.inc

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/**
+ * Participant Portal module.
+ *
+ * PHP Version 7
+ *
+ * @category   Module
+ * @package    Main
+ * @subpackage Participant Portal - View Surveys
+ * @author     Sruthy Mathew <sruthy.mathew@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris-Trunk/
+ */
+namespace LORIS\participant_portal;
+
+/**
+ * Participant Portal class
+ *
+ * PHP Version 7
+ *
+ * @category Module
+ * @package  Main
+ * @author   Sruthy Mathew <sruthy.mathew@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+class View_Surveys extends \NDB_Form
+{
+    /**
+     * Setup function provides the initial data from database
+     *
+     * @return void
+     */
+    function setup(): void
+    {
+
+        $factory = \NDB_Factory::singleton();
+        $baseURL = $factory->settings()->getBaseURL();
+        session_start();
+        $email =$_SESSION['email'];
+        $pid   =$_SESSION['pid'];
+        if ((!isset($email)) && (!isset($pid)) ) {
+            header("Location: $baseURL/participant_portal");
+            exit;
+        }
+        $p = 'page_title';
+        $this->tpl_data[$p] = 'Welcome To Participant Portal! (ID:'.$pid.')';
```

Comment:
`$pid` came directly from the user and is being directly inserted into the smarty template via string concatenation here. This is insecure and vulnerable to an XSS attack.